front/src/reducers/user/index.ts

```diff
@@ -39,12 +44,16 @@ export const userSlice = createSlice({
     logoutSuccess() {
       return userInitialState;
     },
+    setUserRoles(state, action: PayloadAction<{ userRoles: BuiltinRole[] | undefined }>) {
+      state.userRoles = action.payload.userRoles ?? [];
+    },
```

Comment:
You can simplify your action like this
```suggestion
    setUserRoles(state, action: PayloadAction<BuiltinRole[] | undefined>) {
      state.userRoles = action.payload || [];
    },
```